tools/build_langserver/langserver/analyzer.go

```diff
@@ -171,17 +181,17 @@ func newRuleDef(content string, stmt *asp.Statement) *RuleDef {
 				ruleDef.Object = arg.Type[0]
 			} else {
 				// Fill in the ArgMap
-				argString := getArgString(arg)
 				ruleDef.ArgMap[arg.Name] = &Argument{
 					Argument:   &arg,
-					definition: argString,
-					required:   arg.Value == nil,
+					Definition: getArgString(arg),
+					Required:   arg.Value == nil,
 				}
 			}
 		}
 	}
 
-	ruleDef.Header = strings.TrimSuffix(strings.Join(headerSlice, "\n"), ":")
+	header := strings.TrimSuffix(strings.Join(headerSlice, "\n"), ":")
+	ruleDef.Header = removePrivateArgFromHeader(header)
```

Comment:
here

Reply:
as discussed I like the idea of constructing it in the analyzer...

Reply:
In that case, we can just leave the header as it, as i constructed the Repr here now